### ISSUE TITLE
Caffe2 教程-CIFAR 10问题请教

### ISSUE THREAD

**Issue by Olivert007** [hellozhaozheng/LogoDection#1]
楼主你好，尝试多种方法，找你邮箱，没有成功。暂时只能采用这种方式。复现你博客的实验，代码存在一些错误集问题，希望能够沟通交流。
1、创建 LMDBs 数据结构
`        print("Inserted {} rows".format(count))
        print("\nLMDB saved at " + lmdb_path + "\n\n")
        labels_path.close()`
**labels_path.close()**，显示路径无法找到
![Uploading image.png…]()
没有修改成功，代码采用注释，可以生成数据集，无法确实操作是否正确
2、`def AddInput(model, batch_size, db, db_dtype):
    # 加载数据
    data_uint8, label = brew.db_input(
        model, # ModelHelper 实例
        blobs_out=["data_uint8", "label"],
        batch_size=batch_size,
        db=db,
        db_type=db_type,
    )`
*brew*好像没有这个函数`db_input`,代码执行报错，已经解决
3、缺少反向计算函数
`def AddTrainingOperators(model, softmax, label):
    xent = model.LabelCrossEntropy([softmax, label], 'xent')
    # 计算 loss 的期望值
    loss = model.AveragedLoss(xent, "loss")
    # from caffe2.python import optimizer
    optimizer.build_sgd(
        model,
        base_learning_rate=0.01,
        policy="fixed",
        momentum=0.9,
        weight_decay=0.004
    )`
**loss = model.AveragedLoss(xent, "loss")**需要将loss值输入反向计算函数
4、路径拼接操作有误
`import datetime

# 在 root_folder 目录下创建唯一的命名字典
unique_timestamp = str(datetime.datetime.now().strftime('%Y-%m-%d_%H-%M-%S'))
checkpoint_dir = os.path.join(root_folder. unique_timestamp)
`
修改为逗号，已经解决
5、Checkpoint执行文件为空
`# 添加模型核心定义, 将返回的值保存在 'softmax' 变量中
softmax = Add_Original_CIFAR10_Model(
                train_model, data, num_classes,
                image_height, image_width, image_channels
)
# 利用上面返回的 softmax 添加 training operators
AddTrainingOperators(train_model, softmax, label)
# 添加 periodic checkpoint 输出到模型中
AddCheckpoints(train_model, checkpoint_iters, db_type="lmdb")`
检查生成的checkpoint文件是空文件
6、保存训练好的模型是bug，无法继续执行下去
也是最想跟楼主交流的，希望收到你的回复。

**Comment by Olivert007** [comment]
回过头来看原来的问题，发现大家都是抄，还想需要努力啊
